src/languages/asciidoc.js

```diff
@@ -107,18 +107,38 @@ export default function(hljs) {
         begin: '^(NOTE|TIP|IMPORTANT|WARNING|CAUTION):\\s+',
         relevance: 10
       },
-      // inline strong
+      // inline unconstrained strong
+      {
+        className: 'strong',
+        begin: /\*{2}[\w\s\n]+/,
+        end: /\*{2}/
+      },
+      // escaped formatting marks
+      {
+        begin: /\\/,
+        contains: [{
+          begin: /[*_`]/,
+        }]
+      },
+      // inline constrained strong
       {
         className: 'strong',
         // must not follow a word character or be followed by an asterisk or space
-        begin: '\\B\\*(?![\\*\\s])',
-        end: '(\\n{2}|\\*)',
+        // must not be preceded by the escape character \
+        begin: /\B\*(\w\n?\w)+(?!\n\n)/,
```

Comment:
This looks strange.  What is the exact intentional of this rule?  Trying to prevent `**` from wrapping across paragraphs?

Reply:
You are correct, an empty line (two consecutive newline characters) indicates a new paragraph.

_not bold_
```
*bob was

a wild man
```

vs

_bold_
```
*bob was
a wild man*
```

The last example is equivalent to:

```
*bob was a wild man*
```

Reply:
One note here then I'll shut up until you ping me.  We also have a Slack now if that might interest you. (see top of README).

```
       begin: /\B\*(\w\n?\w)+(?!\n\n)/,
```

This is probably way to narrow (and on the other end grabbing way to much).  I presume that other things can exist inside bold - like links, italics, other inline parts of the grammar.  So this rule would fail to take any of that into account (from what I can see).  Plus if you keep extending it to match more and more you run the risk of it matching things that aren't really the "begin" and "end" - which means because this rule ate them, nothing else can match them.  IE, optimally begin and end should "eat"/match as little as possible.  Usually (for begin) this is accomplished with look-ahead.

What you really want here (I think) is:

```ts
begin: /\B\*/,
unlessMiddleContains: /\n\n/ // sadly, this doesn't exist
contains: [ // ... ],
end : /\*/
```

We have `illegal`, but that will just abort.  We can't "backtrack".  So usually what you do (in simpler cases is try to cover the WHOLE expression with a lookahead):

```ts
begin: concat(/\B\*/, lookahead(contentWithoutAsteriskOrPairOfNewlinesEndingInAsterisk)),
contains: [ //... ],
end : /\*/
```

IE, your begin does all the work... I'm just not sure what that regex might look like off the top of my head. :-)

Reply:
```
 begin: /\B\*(\w\n?\w)+(?!\n\n)/,
```

Also since you have no way to pin this I'm pretty sure it will just rewind:

```
*this is bold
but you'd think

that it should not be*
```

[I believe] the regex will rush forward to `think`, see the new paragraph (not a match!) and then just rewind to `bold\n`, which is a match, so the bold rule is a match.